python/ray/_raylet.pyx

```diff
@@ -1056,3 +997,76 @@ cdef class CoreWorker:
             CObjectID c_object_id = object_id.native()
         # Note: faster to not release GIL for short-running op.
         self.core_worker.get().RemoveActiveObjectID(c_object_id)
+
+    # TODO: handle noreturn better
+    cdef store_task_outputs(
+            self, worker, outputs, const c_vector[CObjectID] return_ids,
+            c_vector[shared_ptr[CRayObject]] *returns):
+        cdef:
+            c_vector[size_t] data_sizes
+            c_string metadata_str
+            shared_ptr[CBuffer] empty_metadata
+            c_vector[shared_ptr[CBuffer]] metadatas
+
+        if return_ids.size() == 0:
+            return
+
+        serialized_objects = []
+        for i in range(len(outputs)):
+            return_id, output = return_ids[i], outputs[i]
+            if isinstance(output, ray.actor.ActorHandle):
+                raise Exception("Returning an actor handle from a remote "
+                                "function is not allowed).")
+            elif output is NoReturn:
+                serialized_objects.append(output)
+                data_sizes.push_back(0)
+                metadatas.push_back(empty_metadata)
+            elif isinstance(output, bytes):
+                serialized_objects.append(output)
+                data_sizes.push_back(len(output))
+                metadata_str = RAW_BUFFER_METADATA
+                metadatas.push_back(dynamic_pointer_cast[
+                    CBuffer, LocalMemoryBuffer](
+                    make_shared[LocalMemoryBuffer](
+                        <uint8_t*>(metadata_str.data()), metadata_str.size())))
```

Comment:
Isn't this referring to the memory of a temporary object `metadata_str` which might be deleted after the current iteration of the loop?